workflow/controller/operator.go

```diff
@@ -259,16 +259,15 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 			return
 		}
 		woc.updated = wfUpdate
-		if !acquired {
-			if !woc.releaseLocksForPendingShuttingdownWfs(ctx) {
-				woc.log.Warn("Workflow processing has been postponed due to concurrency limit")
-				phase := woc.wf.Status.Phase
-				if phase == wfv1.WorkflowUnknown {
-					phase = wfv1.WorkflowPending
-				}
-				woc.markWorkflowPhase(ctx, phase, msg)
-				return
-			}
+		if !acquired && !woc.releaseLocksForPendingShuttingdownWfs(ctx) {
+			woc.log.Warn("Workflow processing has been postponed due to concurrency limit")
+			phase := woc.wf.Status.Phase
+			if phase == wfv1.WorkflowUnknown {
+				phase = wfv1.WorkflowPending
+				setWfPodNamesAnnotation(woc.wf)
```

Comment:
hate to be picky but wondering if we could just set this in one place at the top of the function? like maybe logic like "if annotation isn't set, then set it". Is there any reason it should ever not be set? That would handle any future cases where we preemptively exit for whatever reason.